test/unit/test_disco_vpc.py

```diff
@@ -134,6 +134,59 @@ def _create_vpc_mock(CidrBlock):
         possible_vpcs = ['10.0.0.0/26', '10.0.0.64/26', '10.0.0.128/26', '10.0.0.192/26']
         self.assertIn(str(auto_vpc.vpc['CidrBlock']), possible_vpcs)
 
+    @patch('disco_aws_automation.disco_vpc.DiscoRDS')
+    @patch('disco_aws_automation.disco_vpc.DiscoSNS')
+    @patch('disco_aws_automation.disco_vpc.DiscoVPCEndpoints')
+    @patch('disco_aws_automation.disco_vpc.DiscoVPC.config', new_callable=PropertyMock)
+    @patch('boto3.client')
+    @patch('boto3.resource')
+    def test_create_vpc_with_custom_tags(self, boto3_resource_mock, boto3_client_mock, config_mock,
+                                         endpoints_mock, sns_mock, rds_mock):
+        """Test creating a VPC with a dynamic ip range and tags"""
+        # FIXME This needs to mock way too many things. DiscoVPC needs to be refactored
+
+        config_mock.return_value = get_mock_config({
+            'envtype:auto-vpc-type': {
+                'ip_space': '10.0.0.0/24',
+                'vpc_cidr_size': '26',
+                'intranet_cidr': 'auto',
+                'tunnel_cidr': 'auto',
+                'dmz_cidr': 'auto',
+                'maintenance_cidr': 'auto',
+                'ntp_server': '10.0.0.5'
+            }
+        })
+
+        # pylint: disable=C0103
+        def _create_vpc_mock(CidrBlock):
+            return {'Vpc': {'CidrBlock': CidrBlock,
+                            'VpcId': 'mock_vpc_id',
+                            'DhcpOptionsId': 'mock_dhcp_options_id'}}
+
+        # client_mock = MagicMock()
+        # client_mock.create_vpc.side_effect = _create_vpc_mock
+        # client_mock.get_all_zones.return_value = [MagicMock()]
+        # client_mock.describe_dhcp_options.return_value = {'DhcpOptions': [MagicMock()]}
+        # boto3_client_mock.return_value = client_mock
+
+        client_mock = MagicMock()
+        client_mock.create_vpc.side_effect = _create_vpc_mock
+        boto3_client_mock.return_value = client_mock
+
+        resource_mock = MagicMock()
+        resource_mock.Vpc.create_tags.return_value = []
+        boto3_resource_mock.return_value = resource_mock
+
+        my_tags_options = ["productline:astronauts", "mytag:tag_value"]
+        DiscoVPC._get_vpc_cidr = MagicMock()
+        DiscoVPC._get_vpc_cidr.return_value = '10.0.0.0/26'
+        with patch("disco_aws_automation.DiscoVPC._create_new_meta_networks",
+                   return_value=MagicMock(return_value={})):
+            with patch("disco_aws_automation.DiscoVPC._update_dhcp_options", return_value=None):
+                auto_vpc = DiscoVPC('auto-vpc', 'auto-vpc-type', vpc_tags=my_tags_options)
+                possible_vpcs = ['10.0.0.0/26', '10.0.0.64/26', '10.0.0.128/26', '10.0.0.192/26']
+                self.assertIn(str(auto_vpc.vpc['CidrBlock']), possible_vpcs)
```

Comment:
This probably doesn't need to be testing the VPC CIDRs. We already test that in another unit test